File: solutions/python/N-ary Tree Postorder Traversal.py

# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children


class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        res=[]
        self.helper(root,res);
        return res;
    def helper(self,root,arr):
        if root:
            for i in range(0,len(root.children)):
                self.helper(root.children[i],arr)
            arr.append(root.val)

File: solutions/Javascript/N-ary Tree Postorder Traversal.js
function Node(val, children) {
  this.val = val;
  this.children = children;
}

/**
 * @param {Node|null} root
 * @return {number[]}
 */
var postorder = function (root) {
  let res = [];
  helper(root, res);
  return res;
};

function helper(root, arr) {
  if (root === null) {
    return;
  } else {
    if (root.children.length > 0) {
      for (let i = 0; i < root.children.length; i++) {
        helper(root.children[i], arr);
      }
    }
    arr.push(root.val);
  }
}
